vlad_task3/src/ClientProxyObject.cpp

```diff
@@ -58,13 +60,50 @@ namespace client
         return getValue();
     }
 
-    void ClientProxyObject::printHelp()
+    int ClientProxyObject::connect()
     {
-        std::scoped_lock<std::mutex> th_lk(thread_mutex);
-        boost::interprocess::scoped_lock<boost::interprocess::named_mutex> lk(*mutex);
-        int code = HELP;
+        boost::interprocess::scoped_lock<boost::interprocess::named_mutex> proc_lk(*proc_mutex);
+        boost::interprocess::scoped_lock<boost::interprocess::interprocess_mutex> mem_lk(*mem_mutex);
+        int code = ID;
+        setId(id);
         mq->send(&code, sizeof(int), 0);
-        cond->wait(lk);
+        id_cond->wait(mem_lk);
+        if (!getServerAvailable())
+        {
+            throw std::runtime_error("Server is not available");
+        }
+        int clientId = getId();
```

Comment:
Basically here the client is waiting for an id from the server which is pretty complicated.
Wouldn't be easier to add the client id to the send command(please check the other colleague work)?